packages/@cdktf/hcl2cdk/lib/types.ts

```diff
@@ -10,5 +10,12 @@ export type Scope = {
   providerSchema: ProviderSchema;
   providerGenerator: Record<string, TerraformProviderGenerator>;
   constructs: Set<string>;
-  variables: Record<string, { resource: string; variableName: string }>;
+  variables: Record<
+    string,
+    {
+      resource: string;
+      variableName: string;
+    }
+  >;
+  hasTokenBasedTypeCoercion: boolean;
```

Comment:
maybe add a comment that this is just for adding the imports and can be removed as soon as we derive that information from the AST?

Reply:
Else it might be around for longer 😄

Reply:
It might be around for longer, if we decide against doing the refactoring :D But yeah I can add a comment